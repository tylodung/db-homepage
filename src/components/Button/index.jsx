import React from "react"
import PropTypes from "prop-types"

import styles from "./index.module.css"

const Button = props => (
  <button className={styles.button} {...props}>
    { props.children }
  </button>
)

Button.propTypes = {
  type: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
}

Button.defaultProps = {
  type: `button`,
}

export default Button
