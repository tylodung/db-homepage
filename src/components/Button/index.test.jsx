import React from "react"
import { shallow } from "enzyme"
import Button from "./"

describe(`Button`, () => {
  let props
  let mountedComponent
  const wrapper = () => {
    if (!mountedComponent) {
      mountedComponent = shallow(<Button {...props}>Submit</Button>)
    }
    return mountedComponent
  }

  beforeEach(() => {
    props = {
      type: `submit`,
    }
    mountedComponent = undefined
  })

  it(`renders a submit Button correctly`, () => {
    expect(wrapper()).toMatchSnapshot()
  })
})
