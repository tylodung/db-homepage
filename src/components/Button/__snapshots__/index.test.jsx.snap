// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders a submit Button correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    type="submit"
  >
    Submit
  </Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Submit",
      "className": "button",
      "type": "submit",
    },
    "ref": null,
    "rendered": "Submit",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Submit",
        "className": "button",
        "type": "submit",
      },
      "ref": null,
      "rendered": "Submit",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
